### PR TITLE
Document how to ignore mutations matching the source code by regular expression

### PR DESCRIPTION
**Pull request by maks-rafalko**
Docs for https://github.com/infection/infection/pull/1326

### CODE REVIEW

File: src/guide/how-to.md [hunk 1]
```diff
@@ -129,3 +129,86 @@ Line of the code: `App\Api\Product::product::33`
...
+        "MethodRemoval": {
+            "ignoreSourceCodeByRegex": [
+                "Assert::numeric\\(\\$string\\);"
+            ]
+        }
+    }
+}
+```
+
+Ignore any mutants with particular method name
```
Comment (sanmai) [inline src/guide/how-to.md:191]:
```suggestion
Ignore any mutants with particular method name, e.g.:
```

File: src/guide/how-to.md [hunk 2]
```diff
@@ -129,3 +129,86 @@ Line of the code: `App\Api\Product::product::33`
...
+```diff
+- public function methodCall() {
++ protected function methodCall() {
+```
+
+```diff
+- $this->methodCall();
+```
+
+with the following config
```
Comment (sanmai) [inline src/guide/how-to.md:202]:
```suggestion
with the following config:
```

File: src/guide/how-to.md [hunk 3]
```diff
@@ -129,3 +129,86 @@ Line of the code: `App\Api\Product::product::33`
...
+{
+    "mutators": {
+        "global-ignoreSourceCodeByRegex": [
+            ".*methodCall.*"
+        ]
+    }
+}
+```
+
+<p class="tip">Do not add delimiters (`/ /`) to a regular expression - they will be escaped.</p>
```
Comment (sanmai) [inline src/guide/how-to.md:214]:
```suggestion
<p class="tip">Do not add any delimiters (like `/`) to the regular expression: we are adding and escaping them for you.</p>
```

**Review by sanmai — APPROVED** [review]

**Comment by maks-rafalko** [comment]
Thank you @sanmai

**Pull request merged**